src/libLookup/Lookup.cpp

```diff
@@ -5436,145 +5441,193 @@ void Lookup::RectifyTxnShardMap(const uint32_t oldNumShards,
-      {
-        lock_guard<mutex> g(m_mediator.m_mutexDSCommittee);
+    LOG_GENERAL(INFO, "[DSMB]"
+                          << " Sent DS the txns");
 
-        if (m_mediator.m_DSCommittee->empty()) {
-          continue;
-        }
+    DeleteTxnShardMap(shardId);
+  }
+}
 
-        // Send to NUM_NODES_TO_SEND_LOOKUP which including DS leader
-        PairOfNode dsLeader;
-        if (Node::GetDSLeader(m_mediator.m_blocklinkchain.GetLatestBlockLink(),
-                              m_mediator.m_dsBlockChain.GetLastBlock(),
-                              *m_mediator.m_DSCommittee, dsLeader)) {
-          toSend.push_back(dsLeader.second);
-        }
+void Lookup::SendTxnPacketToDS(const uint32_t oldNumShards,
+                               const uint32_t newNumShards) {
+  LOG_MARKER();
 
-        for (auto const& i : *m_mediator.m_DSCommittee) {
-          if (toSend.size() < NUM_NODES_TO_SEND_LOOKUP &&
-              i.second != dsLeader.second) {
-            toSend.push_back(i.second);
-          }
+  SendTxnPacketPrepare(oldNumShards, newNumShards);
 
-          if (toSend.size() >= NUM_NODES_TO_SEND_LOOKUP) {
-            break;
-          }
-        }
-      }
+  SendTxnPacketToShard(newNumShards, true);
+}
+
+void Lookup::SendTxnPacketPrepare(const uint32_t oldNumShards,
+                                  const uint32_t newNumShards) {
+  if (!LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING,
+                "Lookup::SendTxnPacketPrepare not expected to be called from "
+                "other than the LookUp node.");
+    return;
+  }
 
-      P2PComm::GetInstance().SendBroadcastMessage(toSend, msg);
+  if (m_mediator.m_disableTxns) {
+    LOG_GENERAL(INFO, "Txns disabled - skipping dispatch to shards");
+    return;
+  }
 
-      LOG_GENERAL(INFO, "[DSMB]"
-                            << " Sent DS the txns");
+  const uint32_t numShards = newNumShards;
+
+  // map<uint32_t, vector<Transaction>> mp;
+  {
+    lock_guard<mutex> g(m_txnShardMapGeneratedMutex);
+    m_txnShardMapGenerated.clear();
 
-      DeleteTxnShardMap(i);
+    if (!GenTxnToSend(NUM_TXN_TO_SEND_PER_ACCOUNT, m_txnShardMapGenerated,
+                      numShards)) {
+      LOG_GENERAL(WARNING, "GenTxnToSend failed");
+      // return;
     }
   }
+
+  if (oldNumShards != newNumShards) {
+    auto rectifyFunc = [this, oldNumShards, newNumShards]() mutable -> void {
+      RectifyTxnShardMap(oldNumShards, newNumShards);
+    };
+    DetachedFunction(1, rectifyFunc);
+  }
+
+  this_thread::sleep_for(
+      chrono::milliseconds(LOOKUP_DELAY_SEND_TXNPACKET_IN_MS));
+}
+
+void Lookup::SendTxnPacketToNodes(const uint32_t oldNumShards,
+                                  const uint32_t newNumShards) {
+  LOG_MARKER();
+
+  if (!LOOKUP_NODE_MODE) {
+    LOG_GENERAL(WARNING,
+                "Lookup::SendTxnPacketToNodes not expected to be called from "
+                "other than the LookUp node.");
+    return;
+  }
+
+  SendTxnPacketToShard(oldNumShards, newNumShards);
```

Comment:
we are sending to DS shard twice??

Reply:
Typo here, should be SendTxnPacketPrepare